### PR TITLE
Add tutorial for installing Python packages

### PR DESCRIPTION
**Pull request by seisman**

### CODE REVIEW

File: source/programming/python.md [hunk 1]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
 >>> quit()
 ```
 
+## 安装 Python 包
+
+Python 语言的一大特色是其功能强大的标准库和第三方包（也称为模块或库）。
```
Comment (core-man) [inline source/programming/python.md:144]:
```suggestion
Python 语言的一大特色是其功能强大的标准库和第三方软件包（也称模块或库）。
```

File: source/programming/python.md [hunk 2]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+Python 解释器内置了所有标准库，所以安装解释器后就可以直接使用标准库。
```
Comment (core-man) [inline source/programming/python.md:145]:
```suggestion
Python 解释器内置了所有标准库，安装解释器后就可以直接使用标准库，
```

File: source/programming/python.md [hunk 3]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+而第三方包，则需要先安装才能使用。
```
Comment (core-man) [inline source/programming/python.md:146]:
```suggestion
而第三方包需要先安装才能使用。
```

File: source/programming/python.md [hunk 4]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+
+:::{admonition} pip、conda 与 mamba
+
+在学习如何安装 Python 包之前，有必要先了解 pip、conda 和 mamba，以及它们之间的
```
Comment (core-man) [inline source/programming/python.md:150]:
```suggestion
学习如何安装 Python 包之前，有必要先了解 `pip`、`conda` 和 `mamba`，以及它们之间的
```

File: source/programming/python.md [hunk 5]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+区别与联系:
+
+[`pip`](https://pip.pypa.io/)
+: `pip` 是 Python 官方提供的包管理器，可以安装 [Python 包索引网站](https://pypi.org/) 上的
+  Python 包，也可以从源码安装 Python 包。
```
Comment (core-man) [inline source/programming/python.md:155]:
```suggestion
  Python 包，也可用于从源码安装 Python 包。
```

File: source/programming/python.md [hunk 6]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+
+[`conda`](https://docs.conda.io/)
+: `conda` 是 Anaconda/Miniconda 提供的包管理器，其功能强大，不仅可以安装 Python 包，
```
Comment (core-man) [inline source/programming/python.md:158]:
```suggestion
: `conda` 是 Anaconda/Miniconda 提供的包管理器，不仅可以安装 Python 包，
```

File: source/programming/python.md [hunk 7]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+  还可以安装其他语言写的包（理论上可以安装任何软件）。同时还可以管理 Python 环境，
```
Comment (core-man) [inline source/programming/python.md:159]:
```suggestion
  还可以安装其他语言写的包（理论上可以安装任何软件）。它的另一个重要功能是管理 Python 环境，
```

File: source/programming/python.md [hunk 8]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+  在一个系统内安装多个不同版本的 Python 解释器或包。
```
Comment (core-man) [inline source/programming/python.md:160]:
```suggestion
  可用于在一个系统内安装多个不同版本的 Python 解释器或包。
```

File: source/programming/python.md [hunk 9]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+
+  `conda` 功能强大，但其最大的缺点就是**慢**。安装软件包前需要先解析软件包
+  之间的版本依赖关系，这一步骤很慢，下载、安装软件包的过程很慢。
```
Comment (core-man) [inline source/programming/python.md:163]:
```suggestion
  之间的版本依赖关系，这一步骤很慢，使得下载、安装软件包的过程很慢。
```

File: source/programming/python.md [hunk 10]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+
+[`mamba`](https://mamba.readthedocs.org/)
+: `mamba` 是 `conda` 的替代品，其核心代码用 C 语言编写，因而解析软件的版本依赖
+  关系非常快，且可以并行下载和安装软件包，大大减少了安装软件的等待时间。
```
Comment (core-man) [inline source/programming/python.md:167]:
```suggestion
: `mamba` 是 `conda` 的替代品，不仅解析软件的版本依赖关系非常快（其核心代码是用 C 语言编写的），
  还可以并行下载、安装软件包，大大减少了安装软件的时间。
```

File: source/programming/python.md [hunk 11]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+  `mamba` 的用法与 `conda` 几乎完全一致，网络上看到的 `conda` 命令，将 `conda`
+  替换为 `mamba` 即可直接执行。
+:::
+
+**推荐使用包管理器 [`mamba`](https://mamba.readthedocs.org/) 安装和管理 Python 包。**
+
+`mamba` 使用的是 `conda` 的配置文件。在使用 `mamba` 前，先对 `conda` 做简单配置：
```
Comment (core-man) [inline source/programming/python.md:174]:
```suggestion
`mamba` 使用的是 `conda` 的配置文件，因而使用 `mamba` 前，需先对 `conda` 做简单配置：
```

File: source/programming/python.md (hunk 9 above)
Comment (seisman) [reply source/programming/python.md:163]: 解决依赖关系和下载/安装包其实是两件事情。

File: source/programming/python.md [hunk 12]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
...
+: `pip` 是 Python 官方提供的包管理器，可以安装 [Python 包索引网站](https://pypi.org/) 上的
+  Python 包，也可用于从源码安装 Python 包。
+
+[`conda`](https://docs.conda.io/)
+: `conda` 是 Anaconda/Miniconda 提供的包管理器，不仅可以安装 Python 包，
+  还可以安装其他语言写的包（理论上可以安装任何软件）。它的另一个重要功能是管理 Python 环境，
+  可用于在一个系统内安装多个不同版本的 Python 解释器或包。
+
+  `conda` 功能强大，但其最大的缺点就是**慢**。安装软件包前需要先解析软件包
+  之间的版本依赖关系，这一步骤很慢，下载、安装软件包的过程很慢。
```
Comment (seisman) [inline source/programming/python.md:163]:
```suggestion
  `conda` 功能强大，但其最大的缺点就是**慢**。安装软件包前解析软件包之间的版本
  依赖关系很慢，下载和安装也很慢。
```

File: source/programming/python.md (hunk 9 above)
Comment (seisman) [reply source/programming/python.md]: Updated in 5fd38f6da42d9297a8c74b8e4377dcb4672cad51.

File: source/programming/python.md [hunk 13]
```diff
@@ -139,14 +139,76 @@ C 语言使用大括号 `{ }` 划分代码块，而 Python 中使用缩进划分
 >>> quit()
 ```
 
+## 安装 Python 包
+
+Python 语言的一大特色是其功能强大的标准库和第三方软件包（也称模块或库）。
+Python 解释器内置了所有标准库，安装解释器后就可以直接使用标准库，
+而第三方包需要先安装才能使用。
+
+:::{admonition} pip、conda 与 mamba
```
Comment (core-man) [inline source/programming/python.md:148]:
```suggestion
:::{admonition} `pip`、`conda` 与 `mamba`
```

**Review by core-man — APPROVED** [review]

**Pull request merged**